### PR TITLE
[Slack, Tenableio, WithSecure, Zoom] Fix LastRun setting in event collector

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6238

## Description
Fixed the LastRun setting to be after sending events to XSIAM in the event collector.
In packs: `Slack, Tenableio, WithSecure, Zoom`

### CODE REVIEW

File: Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py [hunk 1]
```diff
@@ -397,38 +403,41 @@ def main() -> None:  # pragma: no cover
...
+            if run_vulnerabilities_fetch(last_run=last_run, first_fetch=first_fetch,
+                                         vuln_fetch_interval=vuln_fetch_interval):
+                generate_export_uuid(client, first_fetch, last_run, severity)
+
+            vulnerabilities = fetch_vulnerabilities(client, last_run, severity)
+            events, new_last_run = fetch_events_command(client, first_fetch, last_run, max_fetch)
+
+            call_send_events_to_xsiam(events=events, vulnerabilities=vulnerabilities)
+
+            demisto.info(f'Setting new last_runto {new_last_run}')
```
Comment (dantavori) [inline Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py:439]:
```suggestion
            demisto.debug(f'Setting new last_run to {new_last_run}')
```

File: Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py [hunk 2]
```diff
@@ -201,6 +200,13 @@ def insert_type_to_logs(audit_logs: list, vulnerabilities: list):
         log.update({'xsiam_type': 'vulnerability'})
 
 
+def call_send_events_to_xsiam(events, vulnerabilities):
+    """Enhanced and sends events and vulnerabilities to XSIAM"""
+    insert_type_to_logs(audit_logs=events, vulnerabilities=vulnerabilities)
```
Comment (dantavori) [inline Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py:205]: I see this one was called regardless of the `should_push_events` value, can you double check?

**Review by dantavori — APPROVED** [review]
approving, see notes

File: Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py (hunk 2 above)
Comment (yaakovpraisler) [reply Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py:205]: `call_send_events_to_xsiam` is called in `main` only when `should_push_events` is True

File: Packs/WithSecure/Integrations/WithSecureEventCollector/WithSecureEventCollector.py
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/WithSecure/Integrations/WithSecureEventCollector/WithSecureEventCollector.py:1]: It didn't has this option before as well (maybe by design)

File: Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py (hunk 2 above)
Comment (dantavori) [reply Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py:205]: I know, but `insert_type_to_logs` is called anyway

Comment (yaakovpraisler) [reply Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py:205]: fixed

**Pull request merged**